hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/RocksDbPersistentMap.java

```diff
@@ -133,6 +164,15 @@ public V setValue(V value) {
       @Override
       public void close() {
         iterator.close();
+        if (readOptions != null) {
+          readOptions.close();
+        }
+        if (upperBound.isPresent()) {
```

Comment:
nit: I know it is true today that if `upperBound` is present  `upperBoundSlice` will be not-null. But it can change in future. It would be better if you have null check on `upperBoundSlice` instead `upperBound.isPresent()`. Same for `lowerBoundSlice`.

```suggestion
        if (upperBoundSlice != null) {
```

Reply:
done